core/src/main/java/org/apache/metamodel/data/ScalarFunctionRow.java

```diff
@@ -49,8 +51,32 @@ public Object getValue(int index) throws IndexOutOfBoundsException {
             return _row.getValue(index - scalarFunctionCount);
         }
         final SelectItem selectItem = scalarFunctionSelectItems.get(index);
-        final SelectItem selectItemWithoutFunction = selectItem.replaceFunction(null);
-        return selectItem.getScalarFunction().evaluate(_row, selectItem.getFunctionParameters(), selectItemWithoutFunction);
+        if (selectItem.getScalarFunction().equals(FunctionType.CONCAT)) {
```

Comment:
Completely agree. In fact, I think this is completely wrong: we should use the evaluate method of the concat function class instead of doing any logic in here. I'll get rid of this if.

Thank you!